### PR TITLE
New Why Splinter topic

### PR DESCRIPTION
**Pull request by chenette**
This PR adds initial content to explain Splinter's benefits and key concepts.

Signed-off-by: Anne Chenette <chenette@bitwise.io>

### CODE REVIEW

File: docs/concepts/why_splinter.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
+# Why Splinter?
+
+Splinter is a privacy-focused platform for building distributed applications
+with your choice of distributed ledger technology. Splinter's innovative
+architecture provides a **privacy-first networking layer,** support for
```
Comment (ltseeley) [inline docs/concepts/why_splinter.md]: Looks like the comma is bold here

File: docs/concepts/why_splinter.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  add other types of consensus, such as Byzantine Fault Tolerant (BFT) or Proof
+  of Work (PoW).
+
+* **Custom REST APIs** for extending services to external tools. For example,
+  the Scabbard service uses this API to expose endpoints for smart contract
+  submission.
+
+Splinter provides a growing list of optional features that are needed by many
+applications, so that application developers don't have to reimplement this
+common functionality. For example, the application authorization handler manages
```
Comment (ltseeley) [inline docs/concepts/why_splinter.md]: Might want to put "application authorization handler" in bold to visually indicate that it's specific thing in splinter. Same for Biome below.

File: docs/concepts/why_splinter.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+format. Smart contracts allow even more elaborate patterns for data generation
+and updates.
+
+## <a name="bookmark-open-source"></a> Auditable Open Source
+
+Splinter is open source software, licensed under the [Apache License Version
+2.0](https://github.com/Cargill/splinter/blob/master/LICENSE) software license.
+The Splinter source code is freely available and auditable.
+
+
```
Comment (ltseeley) [inline docs/concepts/why_splinter.md]: Some sections end with two blank lines, others with just one; this should be consistent throughout the doc.

File: docs/concepts/why_splinter.md (hunk 1 above)
Comment (chenette) [reply docs/concepts/why_splinter.md]: Fixed.

File: docs/concepts/why_splinter.md (hunk 2 above)
Comment (chenette) [reply docs/concepts/why_splinter.md]: Good idea. Done.

File: docs/concepts/why_splinter.md (hunk 3 above)
Comment (chenette) [reply docs/concepts/why_splinter.md]: Deleted extra blank lines.

**Pull request merged**